addons/hr_homeworking_calendar/static/src/calendar/common/calendar_common_renderer.xml

```diff
@@ -16,37 +16,56 @@
         </t>
     </t>
 
-    <div t-name="hr_homeworking_calendar.MultiCalendarButtonWorkLocation" class="o_worklocation_btn w-100 d-flex align-items-center">
-        <div class="o_homework_multi d-flex gap-1 ms-1">
-            <t t-foreach="Object.keys(worklocation)" t-as="location" t-key="location">
-                <t t-foreach="worklocation[location]" t-as="wl" t-key="wl.id">
-                    <div class="o_homeworking_content wl_action" t-att-data-location="location" t-att-data-id="wl.id" t-att-data-employee="wl.employeeId">
-                        <i t-attf-class="o_homework_content d-flex align-items-center justify-content-center flex-wrap border border-1 rounded-circle me-0 fa {{iconMap[wl.icon] || 'fa-map-marker'}} wl_color_{{wl.colorIndex}}"/>
-                    </div>
-                </t>
-                <span class="fw-bolder" t-esc="worklocation[location][0].title"/>
+    <div t-name="hr_homeworking_calendar.MultiCalendarButtonWorkLocation" class="o_worklocation_btn my-2 opacity-75 w-100 d-flex align-items-center cursor-default">
+        <div t-if="Object.keys(worklocation).length &gt; 0" class="d-flex gap-1 ms-1 flex-wrap w-75">
+            <t t-set="iconClass" t-value="`o_homework_content d-flex align-items-center justify-content-center flex-wrap border border-1 rounded-circle me-0 fa`"/>
+            <t t-if="scale === 'month'">
+                <div class="o_homework_multi d-flex align-items-center gap-1 ms-1 w-100 overflow-hidden">
+                    <t t-foreach="Object.keys(worklocation)" t-as="location" t-key="location">
+                        <t t-foreach="worklocation[location]" t-as="wl" t-key="wl.id">
+                            <div class="o_homeworking_content wl_action" t-att-data-location="location" t-att-data-id="wl.id" t-att-data-employee="wl.employeeId">
+                                <i t-att-title="wl.title" t-attf-class="{{iconClass}} {{iconMap[wl.icon] || 'fa-map-marker'}} wl_color_{{wl.colorIndex}} wl_outline_{{wl.colorIndex}}"/>
+                            </div>
+                        </t>
+                        <span t-if="worklocation[location].length === 1 &amp;&amp; Object.keys(worklocation).length === 1" class="fw-bolder me-3 text-truncate" t-out="worklocation[location][0].title"/>
```

Comment:
Does not seem to work, I do not see any text when my entries have the same location. Also, you can replace `Object.keys(worklocation)` by location I believe, as defined in the foreach